improver/utilities/spatial.py

```diff
@@ -393,6 +393,101 @@ def process(self, cube: Cube) -> Tuple[Cube, Cube]:
         return tuple(gradients)
 
 
+def maximum_within_vicinity(
```

Comment:
These new methods in `spatial` need unit tests

Reply:
Added.